subsys/net/openthread/rpc/client/ot_rpc_ip6.c

```diff
@@ -140,39 +127,33 @@ const otNetifMulticastAddress *otIp6GetMulticastAddresses(otInstance *aInstance)
 otError otIp6SetEnabled(otInstance *aInstance, bool aEnabled)
 {
 	struct nrf_rpc_cbor_ctx ctx;
+	otError error;
 
 	ARG_UNUSED(aInstance);
 
 	NRF_RPC_CBOR_ALLOC(&ot_group, ctx, 1);
+	nrf_rpc_encode_bool(&ctx, aEnabled);
+	nrf_rpc_cbor_cmd_no_err(&ot_group, OT_RPC_CMD_IP6_SET_ENABLED, &ctx, ot_rpc_decode_error,
+				&error);
 
-	if (!zcbor_bool_encode(ctx.zs, &aEnabled)) {
-		NRF_RPC_CBOR_DISCARD(&ot_group, ctx);
-		return OT_ERROR_INVALID_ARGS;
-	}
-
-	nrf_rpc_cbor_cmd_rsp_no_err(&ot_group, OT_RPC_CMD_IP6_SET_ENABLED, &ctx);
-
-	return decode_ot_error(&ctx);
+	return error;
 }
 
 bool otIp6IsEnabled(otInstance *aInstance)
 {
 	struct nrf_rpc_cbor_ctx ctx;
-	bool enabled = false;
-	bool decoded_ok;
+	bool enabled;
 
 	ARG_UNUSED(aInstance);
 
 	NRF_RPC_CBOR_ALLOC(&ot_group, ctx, 0);
 
 	nrf_rpc_cbor_cmd_rsp_no_err(&ot_group, OT_RPC_CMD_IP6_IS_ENABLED, &ctx);
 
-	decoded_ok = zcbor_bool_decode(ctx.zs, &enabled);
-	nrf_rpc_cbor_decoding_done(&ot_group, &ctx);
+	enabled = nrf_rpc_decode_bool(&ctx);
 
-	if (!decoded_ok) {
-		nrf_rpc_err(-EBADMSG, NRF_RPC_ERR_SRC_RECV, &ot_group, OT_RPC_CMD_IP6_IS_ENABLED,
-			    NRF_RPC_PACKET_TYPE_RSP);
+	if (!nrf_rpc_decoding_done_and_check(&ot_group, &ctx)) {
+		ot_rpc_report_decoding_error(OT_RPC_CMD_IP6_IS_ENABLED);
```

Comment:
I think using this function for reporting response decode errors is wrong as the function reports that the issue is with a packet of type `NRF_RPC_PACKET_TYPE_CMD`.

Could we:
1. rename `ot_rpc_report_decoding_error` to `ot_rpc_report_cmd_decoding_error)
2. add another `ot_rpc_report_rsp_decoding_error`
3. make sure the first function is not used to report command decoding errors?